core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -194,7 +194,7 @@ object ConfigCommand extends Config {
    */
   private def preProcessBrokerConfigs(configsToBeAdded: Properties, perBrokerConfig: Boolean) {
     val passwordEncoderConfigs = new Properties
-    passwordEncoderConfigs ++= configsToBeAdded.asScala.filterKeys(_.startsWith("password.encoder."))
+    passwordEncoderConfigs ++= configsToBeAdded.asScala.filterKeys(_.startsWith("password.encoder.")).toMap
```

Comment:
Views have a lot of problems in 2.11 and 2.12 so we generally stay away from them (they were completely rewritten in Scala 2.13). For code like this where performance doesn't matter, it's fine to go with the simple approach.

Reply:
Do you remember what are these issues?
I am aware of GC pressure for long lived "view" objects since they hold on to the underlining collection. I don't think this issue applies here since `filterKeys` creates a "view" which is converted to a strict `Map` and we never store a reference to the "view".

Reply:
There are several bugs, particularly in 2.11. You can search the bug tracker if you like. People working on the standard library don't recommend views, they suggest `iterator` instead. The migration notes for 2.13 also don't mention views as the replacement for the idiom, the example given:

mapValues and filterKeys now return a MapView instead of a Map | kvs.mapValues(f) | kvs.mapValues(f).toMap
-- | -- | --

https://docs.scala-lang.org/overviews/core/collections-migration-213.html

A recent article talks about two issues: https://www.scala-lang.org/blog/2017/11/28/view-based-collections.html

I also recall an article showing that views didn't even perform better than the strict equivalent in many scenarios they were designed for.

Personally, I don't think it's worth the additional complexity (particularly around debugging).

Reply:
Okay. Thanks for the information. The deprecation notice for `filterKeys` says:
```
(Since version 2.13.0) Use .view.filterKeys(f). A future version will include a strict version of this method (for now, .view.filterKeys(p).toMap).
```
Looks like there may be some inconsistency in the documentation. Like you say, in this particular code it doesn't matter much but maybe we should do what you did in a few other places:
```
scalaMap.filter { case (key, value) => predicate(key) }
```

Reply:
OK, I changed `filterKeys` to `filter` here and a few other places.